### PR TITLE
More README's

### PR DESCRIPTION
**Pull request by ValarDragon**
## Description

Adds ark-poly readme, leaves adding the domain section as a TODO

cref: #4 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: poly/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+This crate implements traits and implementations for polynomials, smooth cosets of fields, and FFTs for these domains.
+
+### Polynomials
+
+Provides the following traits:
+
+- [`Polynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L16):
+Polynomial trait, ensuring that all implementers have basic arithmetic operations implemented.
+(`Add, Sub, Zero`, evaluation at a point, degree, etc.)
```
Comment (Pratyush) [inline poly/README.md:11]:
```suggestion
(`Add`, `Sub`, `Zero`, evaluation at a point, degree, etc.)
```

File: poly/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Multivariate Polynomial trait.
+
+There are also three different implementations of polynomials in this crate.
+
+- [`univariate/DensePolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/univariate/dense.rs#L22):
+Dense univariate polynomials, where a degree `d` polynomial is represented by `d + 1` coefficients.
+Implements [`UVPolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L41)
+- [`univariate/SparsePolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/univariate/sparse.rs#L15): Sparse univariate polynomials.
+A polynomial is represented by a list representing all the non-zero terms in the polynomial.
+Should only be used when a polynomial is mostly non-zero.
```
Comment (Pratyush) [inline poly/README.md:25]:
```suggestion
Should only be used when most coefficients of the polynomial are zero.
```

File: poly/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Implements [`Polynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L16).
+- [`multivariate/SparsePolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/multivariate/sparse.rs#L21):
+Sparse multivariate polynomials.
+The polynomial is represented by its non-zero terms.
+
+A struct [`univariate/DenseOrSparsePolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/univariate/mod.rs#L16) is available,
+that allows the user to abstract over the type of univariate polynomial underlying it is.
```
Comment (Pratyush) [inline poly/README.md:32]:
```suggestion
This crate allows provides the [`univariate/DenseOrSparsePolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/univariate/mod.rs#L16) enum, which allows the user to abstract over the type of underlying univariate polynomial (dense or sparse).
```

File: poly/README.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+Provides the following traits:
+
+- [`Polynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L16):
+Polynomial trait, ensuring that all implementers have basic arithmetic operations implemented.
+(`Add`, `Sub`, `Zero`, evaluation at a point, degree, etc.)
+- [`UVPolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L41):
+Univariate Polynomial trait.
+Essentially it defines ways to serialize to/from a polynomial, given that it is univariate.
+- [`MVPolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L59):
+Multivariate Polynomial trait.
```
Comment (Pratyush) [inline poly/README.md:16]:
```suggestion
- [`Polynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L16):
Requires implementors to support common operations on polynomials, such as `Add`, `Sub`, `Zero`, evaluation at a point, degree, etc.
- [`UVPolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L41): Specifies that a `Polynomial` is actually a *univariate* polynomial, and defines methods to serialize to and from the coefficient representation of the polynomial.
- [`MVPolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L59):
Specifies that a `Polynomial` is actually a *multivariate* polynomial.
```

File: poly/README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+There are also three different implementations of polynomials in this crate.
```
Comment (Pratyush) [inline poly/README.md:18]:
```suggestion
The crate also provides structs that implement these traits:
```

File: poly/README.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+- [`univariate/DensePolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/univariate/dense.rs#L22):
+Dense univariate polynomials, where a degree `d` polynomial is represented by `d + 1` coefficients.
+Implements [`UVPolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L41)
```
Comment (Pratyush) [inline poly/README.md:22]:
```suggestion
Represents degree `d` univariate polynomials via a list of `d + 1` coefficients. This struct implements the [`UVPolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L41) trait.
```

File: poly/README.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+- [`univariate/SparsePolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/univariate/sparse.rs#L15): Sparse univariate polynomials.
+A polynomial is represented by a list representing all the non-zero terms in the polynomial.
+Should only be used when most coefficients of the polynomial are zero.
+Implements [`Polynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L16).
+- [`multivariate/SparsePolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/multivariate/sparse.rs#L21):
+Sparse multivariate polynomials.
+The polynomial is represented by its non-zero terms.
+
+This crate allows provides the [`univariate/DenseOrSparsePolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/univariate/mod.rs#L16) enum, which allows the user to abstract over the type of underlying univariate polynomial (dense or sparse).
```
Comment (Pratyush) [inline poly/README.md:31]:
```suggestion
- [`univariate/SparsePolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/univariate/sparse.rs#L15): Represents degree `d` univariate polynomials via a list containing all non-zero monomials. This should only be used when most coefficients of the polynomial are zero. This struct
implements the [`Polynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/mod.rs#L16) trait (but *not* the `UVPolynomial` trait.
- [`multivariate/SparsePolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/multivariate/sparse.rs#L21):
Represents multivariate polynomials via a list containing all non-zero monomials.

This crate also provides the [`univariate/DenseOrSparsePolynomial`](https://github.com/arkworks-rs/algebra/blob/master/poly/src/polynomial/univariate/mod.rs#L16) enum, which allows the user to abstract over the type of underlying univariate polynomial (dense or sparse).
```

File: poly/README.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
+## ark-poly
+
+This crate implements traits and implementations for polynomials, FFT-friendly subsets of a fields (dubbed "domains"), and FFTs for these domains.
```
Comment (Pratyush) [inline poly/README.md:3]:
```suggestion
This crate implements traits and implementations for polynomials, FFT-friendly subsets of a field (dubbed "domains"), and FFTs for these domains.
```

File: poly/README.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+### Polynomials
+
+The polynomial module provides the following traits for defining polynomials:
```
Comment (Pratyush) [inline poly/README.md:7]:
```suggestion
The `polynomial` module provides the following traits for defining polynomials:
```

**Comment by Pratyush** [comment]
LGTM modulo the last couple of comments

**Comment by ValarDragon** [comment]
LGTM as well

**Review by Pratyush — APPROVED** [review]

**Pull request merged**